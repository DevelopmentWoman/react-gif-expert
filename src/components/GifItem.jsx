import PropTypes from 'prop-types'

export const GifItem = ({title,url})=>{

  // let {title, id} = image
  console.log(title)

  return (
    <div ClassName="card">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  )
}


GifItem.PropTypes={
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}


