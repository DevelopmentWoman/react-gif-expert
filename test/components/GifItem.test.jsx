import {render, screen} from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Probando GifItem', ()=>{
    const title = 'Saitama';
    const url = 'https://one-punch.com/saitana.jpg';
    test('Esta prueba debe hacer match con su snapshot',()=>{

        const {container} = render(<GifItem title={title} url={url}/>)
        expect(container).toMatchSnapshot();
    })

    test('debe mostrar la imagen con el url y el alt indicado', ()=>{
        render(<GifItem title={title} url={url}/>)
        const {src, alt} = screen.getByRole('img')
        expect(src).toBe(url);
        expect(alt).toBe(title);
    })

    test('Debe mostrar el titulo en el componente',()=>{
        render(<GifItem title={title} url={url}/>)
        expect(screen.getByText(title)).toBeTruthy();
    })
})