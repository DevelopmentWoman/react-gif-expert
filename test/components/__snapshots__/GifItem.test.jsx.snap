// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando GifItem Esta prueba debe hacer match con su snapshot 1`] = `
<div>
  <div
    classname="card"
  >
    <img
      alt="Saitama"
      src="https://one-punch.com/saitana.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
